domain/src/main/java/camp/nextstep/edu/domain/calculator/Expression.kt

```diff
@@ -0,0 +1,61 @@
+package camp.nextstep.edu.domain.calculator
+
+
+class Expression(
+    private val thunk: (String) -> Unit
+) {
+
+    private val expressions = mutableListOf<ExpressionItem>()
```

Comment:
```suggestion
    private val _expressions = mutableListOf<ExpressionItem>()
    val expressions: List<ExpressionItem>
        get() = _expressions.toList()
```

get 함수는 식을 반환해주는 것 뿐만 아니라 식을 초기화하는 역할까지 같이 가지고 있네요!
이 경우에는 함수의 역할을 나누어주는 것이 좋을 것 같습니다~
참고: [단일책임의 원칙](https://blog.itcode.dev/posts/2021/08/13/single-responsibility-principle)